### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by j8r**
Improve the CI by adding a format and warning checks.
Fix also a spec not running in the GitHub Actions runner.

### CODE REVIEW

**Comment by j8r** [comment]
I haven't an IOCTL working inside the CI, yet. I put a `pending`.

**Comment by postmodern** [comment]
I think we should pick a more general `ioctl`. How about `SIOCGSTAMP`?
https://manpages.ubuntu.com/manpages/bionic/man2/ioctl_list.2.html

File: spec/ioctl_spec.cr [hunk 1]
```diff
@@ -4,7 +4,7 @@ require "spec"
 describe IOCTL do
   describe ".ioctl" do
     context "when given a valid ioctl request number" do
-      it "must populate the given pointer" do
+      pending "must populate the given pointer" do
```
Comment (postmodern) [inline spec/ioctl_spec.cr:7]: Technically the test works if you run it locally. Could we somehow skip the test only when GitHub CI is detected? I'm sure there's a `GITHUB_` environment variable we could check for?

**Review by postmodern — CHANGES_REQUESTED** [review]
Either we should dynamically skip this test only when GitHub CI is detected (the test passes when ran in a real terminal), or we should pick a more generic `ioctl` to call.

File: spec/ioctl_spec.cr (hunk 1 above)
Comment (j8r) [reply spec/ioctl_spec.cr:7]: Yes I thought about that too.
The best would be to find one `ioctl` call working in the CI environmment.

**Comment by postmodern** [comment]
Ah, actually the `FIO` `ioctl`s are generic `ioctl`s that can be called on any file descriptor. Those should work in CI.
https://man.openbsd.org/ioctl.2

**Comment by postmodern** [comment]
Hmm, I tried testing this, and you need to create a socket, but I'm only getting back `0`.

```crystal
require "socket"
require "ioctl"

FIOSETOWN       = 0x8901_u32
SIOCSPGRP       = 0x8902_u32
FIOGETOWN       = 0x8903_u32
SIOCGPGRP       = 0x8904_u32
SIOCATMARK      = 0x8905_u32
SIOCGSTAMP_OLD  = 0x8906_u32      # Get stamp (timeval)
SIOCGSTAMPNS_OLD = 0x8907_u32     # Get stamp (timespec)

value : Int32 = 0
socket = TCPSocket.new("example.com",80)

IOCTL.ioctl(socket.fd, FIOGETOWN, pointerof(value))

puts value
```

**Comment by j8r** [comment]
I've finally found a simple ioctl that works. Even if an integer passed as an argument returns `0`, what's important is to have successful `ioctl` call.

**Pull request merged**